packages/nodes-base/nodes/Telegram/Telegram.node.ts

```diff
@@ -2187,19 +2176,27 @@ export class Telegram implements INodeType {
 					const dataBuffer = await this.helpers.getBinaryDataBuffer(i, binaryPropertyName);
 					const propertyName = getPropertyName(operation);
 
+					const fileNameUI = this.getNodeParameter('fileName',0) as string;
+					const filename = fileNameUI || binaryData.fileName?.toString();
+
 					body.disable_notification = body.disable_notification?.toString() || 'false';
 
 					const formData = {
 						...body,
 						[propertyName]: {
 							value: dataBuffer,
 							options: {
-								filename: binaryData.fileName,
+								filename,
 								contentType: binaryData.mimeType,
 							},
 						},
 					};
-					responseData = await apiRequest.call(this, requestMethod, endpoint, {}, qs, { formData });
+
+					const formDataOptions = new FormData(formData);
+
+					const options = {headers: formDataOptions.getHeaders(), formData};
+
+					responseData = await apiRequest.call(this, requestMethod, endpoint, {}, qs, options);
```

Comment:
@tahasonmez 
Can you please explain those changes to how formData is passed? Isn't It working as it was before?